src/parser/commands.cpp

```diff
@@ -1586,7 +1586,18 @@ void GetProofCommand::invoke(cvc5::Solver* solver, SymManager* sm)
       {
         ss << "(!" << std::endl;
       }
-      ss << solver->proofToString(p, format);
+      // get assertions, and build a map between them and their names
+      std::vector<cvc5::Term> assertions = solver->getAssertions();
```

Comment:
Here, you can just get the entire map you are interested in directly via `sm->getExpressionMap(true)` https://github.com/cvc5/cvc5/blob/main/src/parser/sym_manager.h#L170.
This would avoid having map lookups for each assertion.

Reply:
Nice! I missed this method when looking at the API of the symbol manager.